sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuDeviceManager.scala

```diff
@@ -203,9 +203,26 @@ object GpuDeviceManager extends Logging {
       var init = RmmAllocationMode.CUDA_DEFAULT
       val features = ArrayBuffer[String]()
       if (conf.isPooledMemEnabled) {
-        init = init | RmmAllocationMode.POOL
-        features += "POOLED"
+        init = conf.rmmPool match {
+          case c if "default".equalsIgnoreCase(c) =>
+            features += "POOLED"
+            init | RmmAllocationMode.POOL
+          case c if "arena".equalsIgnoreCase(c) =>
+            features += "ARENA"
+            init | RmmAllocationMode.ARENA
+          case c if "none".equalsIgnoreCase(c) =>
+            // Pooling is disabled.
+            init
+          case c =>
+            logError(s"RMM pool set to '$c' is not supported.")
+            init
```

Comment:
This should throw IllegalArgumentException or something similar, otherwise this is still proceeding (albeit with a log message in an executor log that the user probably will not immediately see).

Reply:
Done.